integration/next_context_test.go

```diff
@@ -75,3 +76,33 @@ func TestSettingTagAndProjectContext(t *testing.T) {
 	tasks = unmarshalTaskArray(t, output)
 	assert.Equal(t, 0, len(tasks), "no tasks within context project:beta +one")
 }
+
+func TestContextFromEnvVar(t *testing.T) {
+	repo, cleanup := makeDstaskRepo(t)
+	defer cleanup()
+
+	program := testCmd(repo)
+
+	output, exiterr, success := program("add", "+one", "+alpha", "one")
+	assertProgramResult(t, output, exiterr, success)
+
+	output, exiterr, success = program("add", "project:beta", "+two", "two")
+	assertProgramResult(t, output, exiterr, success)
+
+	output, exiterr, success = program("context", "project:beta")
+	assertProgramResult(t, output, exiterr, success)
+
+	// override context with an env var
+	unsetEnv := setEnv("DSTASK_CONTEXT", "+one +alpha")
+	//unsetEnv := setEnv("DSTASK_CONTEXT", "+one")
```

Comment:
I can add more unit tests to confirm this, but setting a space-containing string like "+one +alpha" appears to _parse_ correctly, but still fails our integration test.

Setting a string without spaces (just a single tag in this case) works  fine. I kind of expected both to work.

Maybe there's something about env vars in unix I don't understand.

Reply:
Ah this might be because the full string "+one +alpha" is passed as a single argument to `ParseCmdLine`. At this point, our parsing logic assumes that the command line has been tokenized, and interprets this as a single tag.

Reply:
That was the problem. [This standard library function](https://golang.org/pkg/strings/#Fields) did the trick.